source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -118,6 +135,35 @@ const std::string& Config::getRouteFromEntries(Network::Connection& connection)
   return EMPTY_STRING;
 }
 
+const std::string& Config::getWeightedClusterRoute(const uint64_t random_value) {
+  uint64_t selected_value = random_value % total_cluster_weight_;
+  uint64_t begin = 0UL;
+  uint64_t end = 0UL;
+
+  // Find the right cluster to route to based on the interval in which
+  // the selected value falls. The intervals are determined as
+  // [0, cluster1_weight), [cluster1_weight, cluster1_weight+cluster2_weight),..
+  for (const WeightedClusterEntry& cluster : weighted_clusters_) {
```

Comment:
This code is duplicate of the same thing in thrift_proxy. Is there any way to share the code for this algorithm? Maybe a templatized function just for the algorithm, so you don't need to make the entries be a common type?

Reply:
And that is a dupe of code in hcm

Reply:
Addressed in 9dce4f5.